### PR TITLE
log-backup: use conservativer batch strategy

### LINKED ISSUE #14313: pitr restore data fail for "failed: raft entry is too large, region 170, entry size 8616655"
### What version of TiKV are you using?

2.28 nightly
### Steps to reproduce
start log backup
do full backup every sometime
run paypay workload
do pitr restore with latest full backup repeatedly (clean target TiDB eachtime)

### What did you expect?
pitr backup and restore works normal
### What did happened?
```code
failed: raft entry is too large, region 170, entry size 8616655; execute over region id:170 start_key:\"t\\200\\
000\\000\\000\\000\\000\\000\\377e_i\\200\\000\\000\\000\\000\\377\\000\\000\\001\\003\\200\\000\\000\\000\\377\\000
\\031\\014\\336\\000\\000\\000\\000\\373\" end_key:\"t\\200\\000\\000\\000\\000\\000\\000\\377e_i\\200\\000\\000\\00
0\\000\\377\\000\\000\\001\\003\\200\\000\\000\\000\\377\\000\\035O\\237\\000\\000\\000\\000\\373\" region_epoch:<co
nf_ver:1 version:158 > peers:<id:171 store_id:1 >  failed: raft entry is too large, region 170, entry size 8616655;
execute over region id:170 start_key:\"t\\200\\000\\000\\000\\000\\000\\000\\377e_i\\200\\000\\000\\000\\000\\377\\0
00\\000\\001\\003\\200\\000\\000\\000\\377\\000\\031\\014\\336\\000\\000\\000\\000\\373\" end_key:\"t\\200\\000\\000
\\000\\000\\000\\000\\377e_i\\200\\000\\000\\000\\000\\377\\000\\000\\001\\003\\200\\000\\000\\000\\377\\000\\035O\\
237\\000\\000\\000\\000\\373\" region_epoch:<conf_ver:1 version:158 > peers:<id:171 store_id:1 >  failed: raft entry
 is too large, region 170, entry size 8616655; execute over region id:170 start_key:\"t\\200\\000\\000\\000\\000\\00
0\\000\\377e_i\\200\\000\\000\\000\\000\\377\\000\\000\\001\\003\\200\\000\\000\\000\\377\\000\\031\\014\\336\\000\\
000\\000\\000\\373\" end_key:\"t\\200\\000\\000\\000\\000\\000\\000\\377e_i\\200\\000\\000\\000\\000\\377\\000\\000\
\001\\003\\200\\000\\000\\000\\377\\000\\035O\\237\\000\\000\\000\\000\\373\" region_epoch:<conf_ver:1 version:158 >
 peers:<id:171 store_id:1 >  failed: raft entry is too large, region 170, entry size 8616655; execute over region id
:170 start_key:\"t\\200\\000\\000\\000\\000\\000\\000\\377e_i\\200\\000\\000\\000\\000\\377\\000\\000\\001\\003\\200
\\000\\000\\000\\377\\000\\031\\014\\336\\000\\000\\000\\000\\373\" end_key:\"t\\200\\000\\000\\000\\000\\000\\000\\
377e_i\\200\\000\\000\\000\\000\\377\\000\\000\\001\\003\\200\\000\\000\\000\\377\\000\\035O\\237\\000\\000\\000\\00
0\\373\" region_epoch:<conf_ver:1 version:158 > peers:<id:171 store_id:1 >  failed: raft entry is too large, region
170, entry size 8616655\n", "error": "command terminated with exit code 1"} 
```

### PR DESCRIPTION
**Pull request by YuJuncen**
### What is changed and how it works?

Issue Number: Close #14313

What's Changed:
This PR make the default max batch size 1/2 of the max raft command size. (IMO 4 MiB would be a fair choice for batching, even we batch up to 7MiB, perhaps the throughput won't be greatly increased.)

This PR also make the `Collector` try its best to don't make batch greater than `max_raft_cmd_size`. In the past, it fire the batch after noticing the current `unpacked_size` is greater than the max batch size, which make the real batch size will be slightly greater than the real size. After this, it will check the total size BEFORE it add the request.

```commit-message
```

### Check List 

Tests 

- Unit test

### Release note 

```release-note
Fixed a bug that may cause PITR fail when there are too many tiny or huge rows. 
```

### CODE REVIEW

File: src/import/sst_service.rs [hunk 1]
```diff
@@ -180,10 +183,25 @@ impl RequestCollector {
         self.accept(cf, m);
     }
 
+    /// check whether the unpacked size would exceed the max_raft_req_size after
+    /// accepting the modify.
+    fn should_send_batch_before_adding(&self, m: &Modify) -> bool {
+        let message_size = m.size() + WIRE_EXTRA_BYTES;
```
Comment (3pointer) [inline src/import/sst_service.rs:189]: maybe add some comments about the exact size between `Modify` and `Protobuf`

Comment (YuJuncen) [reply src/import/sst_service.rs:189]: See the comment in the `WIRE_EXTRA_BYTES`.

**Review by tonyxuqqi — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Pull request merged**